lnwallet/channel.go

```diff
@@ -3372,14 +3385,32 @@ func (lc *LightningChannel) validateCommitmentSanity(theirLogCounter,
 	ourInitialBalance := commitChain.tip().ourBalance
 	theirInitialBalance := commitChain.tip().theirBalance
 
-	ourBalance, theirBalance, commitWeight, filteredView, err := lc.computeView(
+	ourBalance, theirBalance, commitWeight, feeSiphon, filteredView, err := lc.computeView(
 		view, remoteChain, false,
 	)
 	if err != nil {
 		return err
 	}
 	feePerKw := filteredView.feePerKw
 
+	// For the anchor channelt type, we want to disallow states where the
+	// potential fees siphoned from HTLC timeout transactions are greater
+	// than what is owed to the node in any future states. Since we know
+	// that they must always keep the reserve available for punishment, we
+	// disallow any state where this wouldn't be the case.
+	reserve := lc.channelState.LocalChanCfg.ChanReserve
+	if remoteChain {
+		reserve = lc.channelState.RemoteChanCfg.ChanReserve
+	}
+
+	// TODO(halseth): this limits the number of HTLCs that can be used
```

Comment:
In practice, do you have some rough figures for the impact here? I wonder if we should expose this as a config option to avoid potentially obstructing certain services/use-cases over night with the update.

Reply:
You mean an option to skip the siphon check? That's not a bad idea..

Reply:
I realized that a problem with having the option to skip the fee leak check only works if both parties skip it. Otherwise the receiver will close the channel if the sender has disabled it and violates the invariant.